packages/create-project/src/index.ts

```diff
@@ -1,151 +1,181 @@
+  const resolvedProjectPath = path.resolve(projectPath);
+  const projectName = path.basename(resolvedProjectPath);
+
+  const { valid, problems } = validateNpmName(projectName);
+  if (!valid) {
+    console.error(
+      `Could not create a project called ${chalk.red(`"${projectName}"`)} because of npm naming restrictions:`
+    );
+
+    problems?.forEach((p) => console.error(`    ${chalk.red.bold('*')} ${p}`));
+    process.exit(1);
   }
-}
 
-async function init() {
-  const { _: args, ...opts } = arg({
-    '--use-yarn': Boolean,
-    '--temporal-version': String,
-    '--sample': String,
-  });
-  if (args.length !== 1) {
-    throw new UsageError();
+  let example = program.example;
+
+  // `example` is true when --example is used by itself
+  if (typeof example !== 'string') {
+    const res = await prompts({
+      type: 'text',
+      name: 'example',
+      message: 'Which template would you like to use?',
+      initial: 'hello-world',
+      // validate: (name) => {
+      //   const validation = validateNpmName(path.basename(path.resolve(name)));
+      //   if (validation.valid) {
+      //     return true;
+      //   }
+      //   return 'Invalid project name: ' + validation.problems![0];
+      // },
+    });
+
+    if (typeof res.example === 'string') {
+      example = res.example;
+    }
   }
-  const sample = opts['--sample'] || 'hello-world';
-  await createProject(args[0], !!opts['--use-yarn'], opts['--temporal-version'] || 'latest', sample);
+
+  if (typeof example !== 'string') {
+    console.error();
+    console.error('Please specify which example:');
+    console.error(`  ${chalk.cyan(program.name())} --example ${chalk.green('[name]|[github-url]')}`);
+    console.error();
+    console.error('For example:');
+    console.error(`  ${chalk.cyan(program.name())} --example ${chalk.green('hello-world')}`);
+    console.error();
+    console.error(`Run ${chalk.cyan(`${program.name()} --help`)} to see all options.`);
+    process.exit(1);
+  }
+
+  await createApp({
+    appPath: resolvedProjectPath,
+    useYarn: !!program.useYarn,
+    example: example.trim(),
+    examplePath: typeof program.examplePath === 'string' ? program.examplePath.trim() : undefined,
+  });
 }
 
-init()
-  .then(() => process.exit(0))
-  .catch((err) => {
-    if (err instanceof UsageError) {
-      console.error(
-        `Usage: ${command} [--use-yarn] [--temporal-version VERSION] [--sample hello-world|hello-world-mtls] <packagePath>`
+const update = checkForUpdate(packageJson).catch(() => null);
+
+async function notifyUpdate(): Promise<void> {
+  try {
+    const res = await update;
+    if (res?.latest) {
+      const isYarn = shouldUseYarn();
+
+      console.log();
+      console.log(chalk.yellow.bold('A new version of `@temporalio/create` is available!'));
+      console.log(
+        'You can update by running: ' +
+          chalk.cyan(isYarn ? 'yarn global add @temporalio/create' : 'npm i -g @temporalio/create')
```

Comment:
Does this guarantee that the package gets upgraded?
Also, we never ask users to install this package directly, maybe suggest running with `npx @temporalio/create@latest ...`?

Reply:
Sounds good